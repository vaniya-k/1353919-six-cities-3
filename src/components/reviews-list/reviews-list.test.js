import React from 'react';
import renderer from 'react-test-renderer';
import ReviewsList from './reviews-list.jsx';

const reviews = [
  {
    avaPicUrl: `https://localhost`,
    date: new Date(`December 17, 2019 03:24:00`),
    name: `Bruce`,
    rating: 1,
    text: `I couldn't stand beeing there -- it's like desert. Why would anyone live in such a place? It's beyond my comprehension`
  },
  {
    avaPicUrl: `https://localhost`,
    date: new Date(`December 25, 2019 13:27:00`),
    name: `Clark`,
    rating: 5,
    text: `Gorgeous ambience, so much light and air! You could even play soccer here!`
  }
];

it(`<ReviewList/> should render two reviews`, () => {

  const tree = renderer
    .create(<ReviewsList reviews={reviews}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
