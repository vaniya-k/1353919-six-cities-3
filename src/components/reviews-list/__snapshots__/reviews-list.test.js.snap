// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList/> should render two reviews 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://localhost"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Bruce
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        I couldn't stand beeing there -- it's like desert. Why would anyone live in such a place? It's beyond my comprehension
      </p>
      <time
        className="reviews__time"
        dateTime={2019-12-17T00:24:00.000Z}
      >
        December 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://localhost"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Clark
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Gorgeous ambience, so much light and air! You could even play soccer here!
      </p>
      <time
        className="reviews__time"
        dateTime={2019-12-25T10:27:00.000Z}
      >
        December 2019
      </time>
    </div>
  </li>
</ul>
`;
